packages/blade/src/components/SideNav/SideNav.web.tsx

```diff
@@ -355,6 +355,11 @@ const _SideNav = (
                     onVisibleLevelChange?.({ visibleLevel: 2 });
                   }, L1_EXIT_HOVER_DELAY);
                 }
+                // If L1 is collapsed and not hovered we want to change visible level to 2
+                // This state/edgecase happens when user clicks on a nested nav and it collapses the L1 causing isL1Hovered to be false
+                if (isL1Collapsed && !isL1Hovered) {
```

Comment:
Didn't understand this scenario, Can you share screenrecording? 

The level isn't changing in this scneario no?

Reply:
https://github.com/user-attachments/assets/8c61d595-2062-429d-9f02-d8433ce94209

Reply:
Ohh got it. Thanks